src/cas.rs

```diff
@@ -1,108 +1,93 @@
-use super::annotation::Annotation;
-use super::error::PipelineError;
 use std::collections::HashMap;
 
+use crate::annotation::Annotation;
+use crate::error::{PipelineError, Result};
+
+#[derive(Clone, Debug, Default)]
 pub struct Cas {
-    pub text: String,
-    pub annotations: HashMap<String, Vec<Annotation>>,
+    text: String,
+    annotations: HashMap<String, Vec<Annotation>>,
 }
 
 impl Cas {
-    pub fn new() -> Cas {
-        Cas {
-            text: String::from(""),
+    pub fn new(text: String) -> Self {
+        Self {
+            text,
             annotations: HashMap::new(),
         }
     }
-    pub fn get_covered_text(self: &Self, begin: usize, end: usize) -> &str {
-        let ref_text: &str = self.text.as_str();
-        &ref_text[begin..end]
-    }
-    pub fn get_covered_text_safe(self: &Self, begin: usize, end: usize) -> Result<&str, String> {
-        let len = self.text.len();
-        if len < end {
-            return Err("End value invalid".to_string());
-        }
-        let ref_text: &str = self.text.as_str();
-        Ok(&ref_text[begin..end])
+
+    pub fn text(&self) -> &str {
```

Comment:
I decided a simple getter is probably better here, since you can just slice the string afterwards if needed.